pkg/scanners/scannerv4/convert_test.go

```diff
@@ -288,11 +289,8 @@ func TestComponents(t *testing.T) {
 	}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			cs := components(tc.metadata, tc.report)
-			assert.Len(t, cs, len(tc.expected))
-			for i, got := range cs {
-				assert.Truef(t, tc.expected[i].EqualVT(got), "expected: %+#v\ngot: %+#v", tc.expected[i], got)
-			}
+			got := components(tc.metadata, tc.report)
+			protoassert.SlicesEqual(t, tc.expected, got, fmt.Sprintf("expected: %+#v\ngot: %+#v", tc.expected, got))
```

Comment:
Alternative (not sure pro/con between the two):

```suggestion
			cs := components(tc.metadata, tc.report)
			assert.Len(t, cs, len(tc.expected))
			for i, got := range cs {
				assert.Truef(t, proto.Equal(tc.expected[i], got), "expected: %+#v\ngot: %+#v", tc.expected[i], got)
			}
```

Reply:
At the moment, I think this is ok, though I mentioned offline I do have concerns about the current implementation of the `protoassert` library. I believe https://github.com/stackrox/stackrox/pull/12878 will alleviate these concerns, and the current implementation of this test will be ok